### PR TITLE
feat(intel_powerstat): Add CPU base frequency metric and add support for new platforms

### PR DESCRIPTION
**Pull request by p-zak**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

Introduced multiple changes:
- New `cpu_base_frequency_mhz` metric added (if configured by adding `cpu_base_frequency` to `package_metrics` configuration option).
- New supported platforms added: 
  - **Intel Lakefield**
  - **Intel RocketLake**
  - **Intel AlderLake**
  - **Intel RaptorLake**
  - **Intel MeteorLake**

New metric:
`cpu_base_frequency_mhz` - will expose base frequency of CPU's package (in MHz)

Example of output for new metric:
```
powerstat_package,host=ubuntu,package_id=0 cpu_base_frequency_mhz=2400i 1669118424000000000
```

### CODE REVIEW

File: plugins/inputs/intel_powerstat/intel_powerstat.go [hunk 1]
```diff
@@ -719,7 +824,7 @@ func (p *PowerStat) verifyProcessor() error {
 	allowedProcessorModelsForC1C6 := []int64{0x37, 0x4D, 0x5C, 0x5F, 0x7A, 0x4C, 0x86, 0x96, 0x9C,
 		0x1A, 0x1E, 0x1F, 0x2E, 0x25, 0x2C, 0x2F, 0x2A, 0x2D, 0x3A, 0x3E, 0x4E, 0x5E, 0x55, 0x8E,
 		0x9E, 0x6A, 0x6C, 0x7D, 0x7E, 0x9D, 0x3C, 0x3F, 0x45, 0x46, 0x3D, 0x47, 0x4F, 0x56,
-		0x66, 0x57, 0x85, 0xA5, 0xA6, 0x8F, 0x8C, 0x8D}
+		0x66, 0x57, 0x85, 0xA5, 0xA6, 0x8A, 0x8F, 0x8C, 0x8D, 0xA7, 0x97, 0x9A, 0xBE, 0xB7, 0xBA, 0xBF, 0xAC, 0xAA}
```
Comment (powersj) [inline plugins/inputs/intel_powerstat/intel_powerstat.go:827]: why this change?

**Review by powersj — COMMENTED** [review]
Thanks for this - just two questions :)

File: plugins/inputs/intel_powerstat/intel_powerstat.go (hunk 1 above)
Comment (p-zak) [reply plugins/inputs/intel_powerstat/intel_powerstat.go:827]: Adding support for platforms:

- Intel Lakefield
- Intel RocketLake
- Intel AlderLake
- Intel RaptorLake
- Intel MeteorLake

File: plugins/inputs/intel_powerstat/intel_powerstat.go [hunk 2]
```diff
@@ -654,6 +683,79 @@ func (p *PowerStat) addCPUC0StateResidencyMetric(cpuID string, acc telegraf.Accu
...
+	if !ok {
+		p.Log.Debugf("cannot find cpuInfo for cpu: %s", cpuID)
+		return 0
+	}
+
+	model := cpuInfo.model
+	busClock100 := []int64{0x2A, 0x2D, 0x3A, 0x3E, 0x3C, 0x3F, 0x45, 0x46, 0x3D, 0x47, 0x4F, 0x56, 0x4E, 0x5E, 0x55, 0x8E, 0x9E, 0xA5, 0xA6, 0x66, 0x6A, 0x6C,
+		0x7D, 0x7E, 0x9D, 0x8A, 0xA7, 0x8C, 0x8D, 0x8F, 0x97, 0x9A, 0xBE, 0xB7, 0xBA, 0xBF, 0xAC, 0xAA, 0x5C, 0x5F, 0x7A, 0x86, 0x96, 0x9C, 0x57, 0x85}
+	busClock133 := []int64{0x1E, 0x1F, 0x1A, 0x2E, 0x25, 0x2C, 0x2F, 0x4C}
+	busClockCalculate := []int64{0x37, 0x4D}
```
[the remark this answers is not in the document]
Comment (p-zak) [reply plugins/inputs/intel_powerstat/intel_powerstat.go:727]: We thought about it but for the readability sake we just wanted to have processor models written in hexadecimal format for now.
`getBusClock` function is called only once (during `Init`) so it shouldn't cause any performance issues.

**Review by powersj — APPROVED** [review]

File: plugins/inputs/intel_powerstat/intel_powerstat.go [hunk 3]
```diff
@@ -75,21 +80,46 @@ func (p *PowerStat) Init() error {
...
-	if !p.areCoreMetricsEnabled() && !p.areGlobalMetricsEnabled() {
-		return fmt.Errorf("all configuration options are empty or invalid. Did not find anything to gather")
-	}
+// fill CPUBusClockValue if required
+func (p *PowerStat) fillCPUBusClock() {
+	if p.packageCPUBaseFrequency {
+		// cpuBusClock is the same for every core/socket.
+		busClockInfo := p.getBusClock("0")
+		if busClockInfo == 0 {
+			p.Log.Debug("Disabling package metric: cpu_base_frequency_mhz. Can't detect bus clock value")
```
Comment (srebhan) [inline plugins/inputs/intel_powerstat/intel_powerstat.go:116]: Shouldn't this be a **warning** here as people enable this feature but won't see it? The debug message might go unnoticed for the average Joe...

**Review by srebhan — COMMENTED** [review]
Looks good to me, my only comment is to make the missing of the `cpu_base_frequency` feature a warn instead of a debug message. What do you think?

File: plugins/inputs/intel_powerstat/intel_powerstat.go (hunk 3 above)
Comment (p-zak) [reply plugins/inputs/intel_powerstat/intel_powerstat.go]: @srebhan Changed. 

I don't know what's going on with `run-readme-linter` job, but for this readme result is:
`Pass plugins/inputs/intel_powerstat/README.md`

**Comment by PanKaker** [comment]
Hi, @srebhan!
Is it possible to add me to Co-authored list of the PR/commit to master? 
https://github.com/influxdata/telegraf/pull/12452/commits/1a931cd9747f09649495aa435f5b67ec4d38b1f7

I was a person who implemented this feature, but because of lack CCLA permissions, we've decided to make this PR under p-zak's name.

Thank you

**Comment by srebhan** [comment]
Hey @PanKaker! Sorry for me missing out on mentioning you in the commit. However as we already merged this, adding you would require to edit git history with forced-push which will in turn give all current PR contributors a big headache. 

Please accept my apologies!

**Comment by VolunPotest** [comment]
Okay. Thank you
